tenant/src/main/java/org/killbill/billing/tenant/api/user/DefaultTenantUserApi.java

```diff
@@ -91,6 +91,11 @@ public Tenant createTenant(final TenantData data, final CallContext context) thr
             throw new TenantApiException(ErrorCode.EXTERNAL_KEY_LIMIT_EXCEEDED);
         }
 
+        // Not transactional, but there is a db constraint on that column
+        if (data.getApiKey() != null && getTenantByApiKey(data.getApiKey()) != null) {
+            throw new TenantApiException(ErrorCode.TENANT_ALREADY_EXISTS, data.getExternalKey());
```

Comment:
It seems like we have [such constraint](https://github.com/killbill/killbill/blob/killbill-0.19.6/tenant/src/main/resources/org/killbill/billing/tenant/ddl.sql#L18) though.

Reply:
I just sync'ed up with @pierre and this is is fine -- seems like this is already the model we have in place for other resources.